api/v1alpha1/openapi.yaml

```diff
@@ -2549,26 +2521,169 @@ components:
+          description: |
+            The name of the repository resource to use as the sync source for environment variables
+        targetRevision:
+          type: string
+          description: |
+            The specific revision of the repository to pull from
+        envFilePath:
+          type: string
+          description: |
+            Path to a file within the repository that contains environment variables to be consumed.
+      required:
+        - repository
+        - targetRevision
+        - envFilePath
+    KubernetesSecretRepositoryRef:
+      type: object
+      properties:
+        name:
+          type: string
+        namespace:
+          type: string
+        mountPath:
+          type: string
+      required:
+        - name
+        - namespace
+        - mountPath
+    HttpRepositoryRef:
+      type: object
+      properties:
+        repository:
+          type: string
+          description: |
+            The name of the repository resource to use as the sync source
+        suffix:
+          type: string
+          description: |
+            Part of the URL that comes after the base URL. It can include query parameters such as:
+            /path/to/endpoint?query=param
+        filePath:
+          type: string
+          description: |
+            The path of the file where the response is stored in the filesystem of the device.
+      required:
+        - repository
+        - filePath
+    ApplicationSpec:
+      type: object
+      description: "Defines the specification for an application"
+      properties:
+        name:
+          type: string
+          description: "The unique name of the application"
+        bundle:
+          $ref: '#/components/schemas/ApplicationBundleProviderSpec'
+        envVars:
+          $ref: '#/components/schemas/ApplicationEnvVars'
+        envFiles:
+          type: array
+          description: "Environment variable key-value pairs from env files, injected at runtime."
+          items:
+            $ref: '#/components/schemas/GitEnvRepositoryRef'
+      required:
+        - name
+        - bundle
+    ApplicationEnvVars:
+      type: object
+      description: "Environment variable key-value pairs, injected during runtime"
+      additionalProperties:
+        type: string
+    ApplicationBundleImageRef:
+      type: object
+      properties:
+        imageRef:
+          type: string
+          description: "The OCI image reference that defines the bundle"
+      required:
+        - imageRef
+    RenderedApplicationSpec:
+      type: object
+      properties:
+        bundle:
+          type: object
+          properties:
+            image:
+              $ref: '#/components/schemas/ApplicationBundleImageRef'
+            data:
+              $ref: '#/components/schemas/ApplicationBundle'
+          oneOf:
+            - required: ["image"]
+            - required: ["data"]
+        envVars:
+          $ref: '#/components/schemas/ApplicationEnvVars'
```

Comment:
If an application type offers passing in both envVars and envFiles (e.g. Helm `--set` vs `-f` flags), they typically define an order of precedence for duplicate keys that the service wouldn't know about. Converting to envVars may then cause unexpected behavior.

Reply:
I suppose but that can be solved with documentation. Not shipping a file greatly simplifies delivery.

Reply:
I think we can leave this as is and then followup with file delivery